### PR TITLE
ci: Check MSRV

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
Pull request #28 broke MSRV.

I can bump it here or I can look at reworking things to not require a bump. Let me know.

Also, I've restructured CI in this PR now to handle the additional checks and simplify the CI.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,28 +3,25 @@ on: [push, pull_request]
...
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
+      - uses: dtolnay/rust-toolchain@master
         with:
+          toolchain: ${{ matrix.toolchain }}
           components: clippy
+      # Default features
+      - run: cargo check --all-targets
+      - run: cargo test
       - run: cargo clippy --all-targets -- -D warnings
```
Comment (pyfisch) [inline .github/workflows/ci.yml:23]: I would run clippy just on stable with default features. The rationale is that the clippy warnings are improved between releases and I don't want to fix outdated or conflicting warnings which might occur.

**Comment by pyfisch** [comment]
> I can bump it here or I can look at reworking things to not require a bump. Let me know.

Bump it here to 1.60, thanks.

**Comment by waywardmonkeys** [comment]
Okay, will take care of both when I am home tonight. On a long weekend in Bangkok at the moment.

**Comment by pyfisch** [comment]
Resolved by #33

**Comment by waywardmonkeys** [comment]
Thanks!

**Pull request closed without merging**